tests/samples/expected/bash.sh

```diff
@@ -150,4 +150,3 @@ export xyzzy=$(
   something
   another_thing --foo
 )
-
```

Comment:
Why is the EOF newline deleted here?

Reply:
That's not the EOF new line; it's an additional, trailing new line that was added to make the tests pass:

```console
$  xxd tests/samples/expected/bash.sh | tail -3
00000610: 736f 6d65 7468 696e 670a 2020 616e 6f74  something.  anot
00000620: 6865 725f 7468 696e 6720 2d2d 666f 6f0a  her_thing --foo.
00000630: 290a                                     ).
```

Reply:
I fear I don't understand. I think we want the newline here, in the expected file, don't we?
If anything, we could remove the trailing newline from the input file, and check that it has been added in the expected file.

Reply:
My understanding of #48 is that some rules end up adding leading or trailing blank lines in the output, which are undesirable but difficult to remove using just the query language (e.g., you can't anchor against the start/end of a file, AFAIK), thus post-processing should be used to remove them instead.

In the Bash queries, the output gets a _lot_ of unnecessary blank lines (see #203), including at EOF (not including the final, necessary `\n`). Thus it's common -- as in the integration test case -- to get an output that ends with two `\n`s. This PR, under my (potentially incorrect :sweat_smile:) understanding of #48, strips them off and then adds the final `\n`; hence the change to the expected output to match.

In general, if my source file looks something like this:
```
.
...
XX..
XXX.
X
XX...
.
```
where `X` is content and `.` is trailing whitespace -- either trailing spaces/etc. on a line, or leading/trailing blank lines -- then I think I would _always_ want it transformed into:
```
XX
XXX
X
XX
```
That's what this PR achieves. Previously, the output would have looked like this:
```
.
.
XX
XXX
X
XX
.
```
(i.e., Just the trailing spaces on each line are removed.)

Reply:
I think that, in your example, I would like the output to be
```
XX
XXX
X
XX
.
```
I thought trailing blank lines were desirable (`git` berates me when I forget mine).

Reply:
Here I'm making a distinction between "blank lines" and "new lines". The previous trimming function outputs this:

```
\n
\n
XX\n
XXX\n
X\n
XX\n
\n
```

The change in this PR outputs this:

```
XX\n
XXX\n
X\n
XX\n
```

i.e., The trailing blank line is removed, but the EOF new line is preserved. (Apologies for the confusion!)

Reply:
Alright, everything is fine then!

Reply:
:tada: